app/tests/components_tests/test_views.py

```diff
@@ -539,3 +543,74 @@ def test_interfaces_list_link_in_new_interface_form(
         user=u,
     )
     assert reverse(interface_list_viewname) in response.rendered_content
+
+
+@pytest.mark.parametrize(
+    "base_object_factory,base_obj_lookup,object_factory,viewname",
+    (
+        (
+            ReaderStudyFactory,
+            "reader_study",
+            DisplaySetFactory,
+            "reader-studies:display-set-update",
+        ),
+        (
+            ArchiveFactory,
+            "archive",
+            ArchiveItemFactory,
+            "archives:item-edit",
+        ),
+    ),
+)
+@pytest.mark.django_db
+def test_image_widget_populated_value_on_update_view_validation_error(
+    client, base_object_factory, base_obj_lookup, object_factory, viewname
+):
+    image1 = ImageFactory()
+    image_ci = ComponentInterfaceFactory(kind=ComponentInterface.Kind.IMAGE)
+    image1_civ = ComponentInterfaceValueFactory(
+        interface=image_ci, image=image1
+    )
+
+    annotation = "{}"
+    annotation_ci = ComponentInterfaceFactory(
+        kind=ComponentInterface.Kind.TWO_D_BOUNDING_BOX
+    )
+    annotation_civ = ComponentInterfaceValueFactory(
+        interface=annotation_ci, value=annotation
+    )
+
+    editor = UserFactory()
+    base_obj = base_object_factory()
+    base_obj.add_editor(editor)
+
+    object = object_factory(**{base_obj_lookup: base_obj})
+    object.values.set([image1_civ, annotation_civ])
+
+    image2 = ImageFactory()
+    data = {
+        "interface_slug": image_ci.slug,
+        "current_value": image2.pk,
+    }
+    data.update(
+        **get_interface_form_data(interface_slug=image_ci.slug, data=image2.pk)
+    )
+    data.update(
+        **get_interface_form_data(
+            interface_slug=annotation_ci.slug, data='{"1":1}'
+        )
+    )
+
+    response = get_view_for_user(
+        client=client,
+        viewname=viewname,
+        reverse_kwargs={"slug": base_obj.slug, "pk": object.pk},
+        user=editor,
+        method=client.post,
+        follow=True,
+        data=data,
+    )
+    assert response.status_code == 200
+    assert f'<option value="IMAGE_SELECTED">{image2.title}</option>' in str(
+        response.rendered_content
+    )
```

Comment:
Don't you also want to check that this is a hidden field?

Reply:
The hidden field already tested in test_widgets.py. Here it tests that the form value is set correctly in the image widget drop down upon validation error. Also, the hidden field is asynchronously populated with htmx so does not appear in the view test response.rendered_content.

Reply:
Maybe I'm missing something, but if all you're testing is the widget, it would be more straight forward to test the form directly, rather than go through the view

Reply:
What I am testing is that the widget is correctly populated with values on form redisplay upon submission in create/update views. So, I should delete this test?

Reply:
Discussed in synchronous meeting